tensorflow_addons/seq2seq/tests/attention_wrapper_test.py

```diff
@@ -44,217 +41,193 @@ def setUp(self):
+    loaded_model.compile("rmsprop", "mse")
+
+    y = loaded_model.predict_on_batch([x_test, dummy_data.query, dummy_data.state])
+
+    np.testing.assert_allclose(y_ref, y)
+
+
+@pytest.mark.parametrize("attention_cls", attention_classes)
+def test_manual_memory_reset(attention_cls):
+    dummy_data = DummyData()
+    attention = attention_cls(dummy_data.units)
+
+    def _compute_score(batch_size=None):
+        if batch_size is None:
+            batch_size = dummy_data.batch
+        memory = dummy_data.memory[:batch_size]
+        attention.setup_memory(
+            memory, memory_sequence_length=dummy_data.memory_length[:batch_size]
+        )
+        assert attention.values.shape.as_list() == list(memory.shape)
+        assert attention.keys.shape.as_list() == list(memory.shape)[:-1] + [
+            dummy_data.units
+        ]
+        return attention([dummy_data.query[:batch_size], dummy_data.state[:batch_size]])
+
+    _compute_score(batch_size=dummy_data.batch)
+    variables = list(attention.variables)
+    _compute_score(batch_size=dummy_data.batch - 1)
+
+    # No new variables were created.
+    for var_1, var_2 in zip(variables, list(attention.variables)):
+        assert var_1 is var_2
+
+
+def test_masking():
+    memory = tf.ones([4, 4, 5], dtype=tf.float32)
+    memory_sequence_length = tf.constant([1, 2, 3, 4], dtype=tf.int32)
+    query = tf.ones([4, 5], dtype=tf.float32)
+    state = None
+    attention = wrapper.LuongAttention(5, memory, memory_sequence_length)
+    alignment, _ = attention([query, state])
+    assert np.sum(np.triu(alignment, k=1)) == 0
+
+
+@pytest.mark.parametrize("attention_cls", attention_classes)
+def test_memory_re_setup(attention_cls):
+    class MyModel(tf.keras.models.Model):
+        def __init__(self, vocab, embedding_dim, memory_size, units):
+            super().__init__()
+            self.emb = tf.keras.layers.Embedding(vocab, embedding_dim, mask_zero=True)
+            self.encoder = tf.keras.layers.LSTM(memory_size, return_sequences=True)
+            self.attn_mch = attention_cls(units)
+
+        def call(self, inputs):
+            enc_input, query, state = inputs
+            mask = self.emb.compute_mask(enc_input)
+            enc_input = self.emb(enc_input)
+            enc_output = self.encoder(enc_input, mask=mask)
+            # To ensure manual resetting also works in the graph mode,
+            # we call the attention mechanism twice.
+            self.attn_mch(enc_output, mask=mask, setup_memory=True)
+            self.attn_mch(enc_output, mask=mask, setup_memory=True)
+            score = self.attn_mch([query, state])
+            return score
+
+    vocab = 20
+    embedding_dim = 6
+    num_batches = 5
+
+    dummy_data = DummyData()
+    model = MyModel(vocab, embedding_dim, dummy_data.memory_size, dummy_data.units)
+    if tf.executing_eagerly():
+        model.compile("rmsprop", "mse", run_eagerly=True)
+    else:
         model.compile("rmsprop", "mse")
```

Comment:
Will this logic ever switch? TF would always be running eagerly right? The  `maybe_run_functions_eagerly` fixture wouldn't trip this so instead maybe we parameterize this as a boolean?

Reply:
Good catch, fixed it!